python-manual/modules/ROOT/pages/migration.adoc

```diff
@@ -0,0 +1,333 @@
+`Result.to_df()` allows for exporting a `Result` object into a Pandas DataFrame.
+
+See xref:transformers.adoc#_transform_to_pandas_dataframe[Manipulate query results -> Transform to Pandas Dataframe].
+====
+|
+.Type hints for all public APIs
+[%collapsible]
+====
+All public objects have type annotations via the link:https://docs.python.org/3/library/typing.html[typing] package. For (optional) use with type checkers and linters.
+====
+
+|
+.Re-authentication
+[%collapsible]
+====
+Allows for handling expiring authentication (backwards compatible) as well as session scoped authentication.
+
+See link:https://github.com/neo4j/neo4j-python-driver/discussions/921[Re-authentication].
+====
+|
+.Mutual TLS (mTLS) as second authentication factor (2FA)
+[%collapsible]
+====
+Allows for configuring client side TLS certificates to authenticate against the server.
+
+See link:https://github.com/neo4j/neo4j-python-driver/discussions/1031[Mutual TLS].
+====
+
+|
+.`BookmarkManager` support
+[%collapsible]
+====
+Bookmark managers make it easier to achieve causal chaining of sessions.
+
+See xref:bookmarks.adoc[].
+====
+|
+.Notification filtering API
+[%collapsible]
+====
+Filtering allows to receive only a subset of notifications from the server, and to improve performance server-side.
+
+See xref:result-summary.adoc#_filter_notifications[Explore the query execution summary -> Filter notifications].
+====
+
+|
+.GQL statuses of queries in the ResultSummary
+[%collapsible]
+====
+Eventually, this API is planned to supersede the current notifications API.
+
+See link:https://github.com/neo4j/neo4j-python-driver/discussions/1067[GQL statuses in ResultSummary].
+====
+|
+.More convenient handling of server-side warnings and notifications
+[%collapsible]
+====
+You can configure the driver to emit warnings on notifications and/or to log them.
+
+See link:https://github.com/neo4j/neo4j-python-driver/discussions/1064[Convenient DBMS notifications].
+====
+
+|
+.Telemetry
+[%collapsible]
+====
+The driver sends anonymous API usage statistics to the server if requested.
+Use the driver configuration `telemetry_disabled=True` to opt out.
+====
+|
+.Concurrency misuse checks
+[%collapsible]
+====
+For local development, run your application with `python -X dev ...` or set the environment variable `PYTHONNEO4JDEBUG` to anything non-empty to get additional concurrency misuse checks.
+Some driver primitives (e.g. sessions) are not safe to be used concurrently, and using them so will lead to hard-to-debug errors.
+This feature helps identifying such code paths (detection is still inherently racy, hence not guaranteed to always work).
+====
+
+|===
+
+
+[[breaking-changes]]
+== Breaking changes and deprecations
+
+The column `Assistent` denotes whether the Neo4j Drivers Migration Assistent can detect issues related to the given changelog entry.
+Entries with a cross mark {cross-mark} require manual inspection.
+
+Deprecated features get removed in the next LTS version.
```

Comment:
That's not correct. Some of them will get removed in `6.0`. But possibly not all of them, we might decide to leave some deprecated APIs around for a prolonged migration period, especially if we anticipate much migration pain.

Reply:
"are likely to be removed in the next major release"?